app/models/calagator/event.rb

```diff
@@ -85,6 +85,22 @@ class Event < ActiveRecord::Base
     on_or_after_date(start_date).before_date(end_date)
   }
 
+  scope :within_times, ->(start_time, end_time) {
+    future.order(:start_time).select{|rec|
```

Comment:
Ideally, these time filters would work in concert with the date filters, so we might not want to use `future` here in all cases.